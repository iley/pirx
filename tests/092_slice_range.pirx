func main(): int {
  var s = new([]int, 5);
  s[0] = 10;
  s[1] = 20;
  s[2] = 30;
  s[3] = 40;
  s[4] = 50;

  var r = range(s, 1, 4);
  printf("size = %d\n", getsize(r));
  printf("values = %d %d %d\n", r[0], r[1], r[2]);

  var r2 = range(s, 0, 2);
  printf("r2 size = %d\n", getsize(r2));
  printf("r2 values = %d %d\n", r2[0], r2[1]);

  var str = "Hello, World!";
  var substr = range(str, 7, 12);
  printf("substr size = %d\n", getsize(substr));

  return 0;
}
