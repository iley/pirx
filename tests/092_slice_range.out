size = 3
values = 20 30 40
r2 size = 2
r2 values = 10 20
substr size = 5
